### PR TITLE
Get layer version by Catalog Client.

### PR DESCRIPTION
**Pull request by iriza7**
* Create LayersVersionRequest class.

* Get layers version by Catalog Client.

* Add unit tests.

Relates-To: OLPEDGE-919.

Signed-off-by: Bautista, Iryna <ext-iryna.bautista@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 1]
```diff
@@ -101,6 +102,32 @@ export class CatalogClient {
         return latestVersion.version;
     }
 
+    public async getLayersVersion(
```
[the remark this answers is not in the document]
Comment (iriza7) [reply @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: fixed

File: @here/olp-sdk-dataservice-read/lib/client/LayersVersionRequest.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { validateBillingTag } from "..";
+
+export class LayersVersionRequest {
```
[the remark this answers is not in the document]
Comment (iriza7) [reply @here/olp-sdk-dataservice-read/lib/client/LayersVersionRequest.ts]: fixed

File: @here/olp-sdk-dataservice-read/lib/client/LayersVersionRequest.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    /**
+     * Catalog version.
+     */
+    public getVersion(): number | undefined {
+        return this.version;
+    }
+
+    /**
+     * Set version value to use in methods getVersion from [[CatalogClient]].
```
[the remark this answers is not in the document]
Comment (iriza7) [reply @here/olp-sdk-dataservice-read/lib/client/LayersVersionRequest.ts]: fixed

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 4]
```diff
@@ -101,6 +102,48 @@ export class CatalogClient {
...
+
+        if (!requestedCatalogVersion) {
+            const billingtag = request.getBillingTag();
+            const catalogVersionRequest = new CatalogVersionRequest();
+            if (billingtag) {
+                catalogVersionRequest.withBillingTag(billingtag);
+            }
+            requestedCatalogVersion = await this.getLatestVersion(
+                catalogVersionRequest
+            );
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Please handle a possible error if the getLatestVersion returns error

File: @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts [hunk 5]
```diff
@@ -132,6 +134,33 @@ describe("CatalogClient", () => {
...
+            version: 3
+        };
+
+        getLayerVersionsStub.callsFake(
+            (builder: any, params: any): Promise<MetadataApi.LayerVersions> => {
+                return Promise.resolve(mockedVersion);
+            }
+        );
+
+        const catalogRequest = new dataServiceRead.LayerVersionsRequest().withVersion(
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts]: Please add test case with calling getLayerVersions without set version in the LayerVersionsRequest

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 6]
```diff
@@ -113,6 +114,56 @@ export class CatalogClient {
...
+            }
+            requestedCatalogVersion = await this.getLatestVersion(
+                catalogVersionRequest
+            ).catch(err => {
+                latestVersionError = err;
+                return undefined;
+            });
+
+            if (!requestedCatalogVersion) {
+                return Promise.reject(`Failed to get latest version`);
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: return Promise.reject(`Failed to get the latest version with error: ${latestVersionError}`);

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 7]
```diff
@@ -113,6 +114,56 @@ export class CatalogClient {
...
+            }
+        }
+
+        const layerVersions = await MetadataApi.getLayerVersions(builder, {
+            version: requestedCatalogVersion,
+            billingTag: request.getBillingTag()
+        });
+
+        return Promise.resolve(layerVersions.layerVersions);
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Please catch an error if getLayerVersions return nothing end reject the promise for the user with the caught error message

**Pull request closed without merging**